File: cpp/design-patterns/builder/naive/request_naive.cpp
class HttpRequest {
public:
    HttpRequest(std::string method, std::string url,
                std::map<std::string,std::string> headers = {},
                std::string body = "", int timeoutMs = 0,
                int retries = 0, bool followRedirects = false);
};

// At the call site, nobody can read this:
HttpRequest("POST", "/api/orders", headers, "{}", 0, 3, true);

File: python/design-patterns/builder/naive/request_naive.py
class HttpRequest:
    def __init__(self, method, url, headers=None, body=None,
                 timeout_ms=None, retries=None, follow_redirects=None):
        ...  # assign all seven

# At the call site, nobody can read this:
HttpRequest("POST", "/api/orders", {"Content-Type": "application/json"},
            '{"id":7}', None, 3, True)
#                       ^^^^  ^  ^^^^  what is None? 3? True?